### ISSUE TITLE
beforeMultiline and newlines after =

### ISSUE THREAD

**Issue by Prometheus3375** [scalameta/scalafmt#2820]
**Integration**: IntelliJ

**Configuration**:
```
version = 3.0.7

runner.dialect = "scala212"

newlines {
    source = fold
    topLevelStatementBlankLines = [
        {
            blanks = 1
        }
    ]
    beforeTemplateBodyIfBreakInParentCtors = true
    beforeMultiline = fold
    forceBeforeMultilineAssign = never
    alwaysBeforeElseAfterCurlyIf = true
    beforeCurlyLambdaParams = never
    afterCurlyLambdaParams = never
    implicitParamListModifierPrefer = after
    avoidForSimpleOverflow = [tooLong, slc]
    avoidInResultType = true
    selectChains = unfold
}

maxColumn = 100
```

## Steps

Given code like this:
```scala
object LL {
  val individualEntrepreneurExtended = extendLegal(data("individualEntrepreneur"), initialize_spsources)
    .withColumnRenamed("primaryAddress", "address")
    .drop("establishments")
}
```

Scalafmt formats in the next way:
```scala
object LL {
  val individualEntrepreneurExtended =
    extendLegal(data("individualEntrepreneur"), initialize_spsources)
      .withColumnRenamed("primaryAddress", "address")
      .drop("establishments")
}
```

With changed `beforeMultiline` to `unfold`:
```scala
object LL {
  val individualEntrepreneurExtended = extendLegal(
    data("individualEntrepreneur"),
    initialize_spsources,
  ).withColumnRenamed("primaryAddress", "address").drop("establishments")
}
```

## Expectation

I expect that with `beforeMultiline = fold` there will be no newline after `=` and the whole formatting will be the same as with `unfold` setting.

For example:
```scala
object LL {
  val scalapbRuntime = "com.thesamet.scalapb" %% "scalapb-runtime" % scalapb.compiler.Version.scalapbVersion % "protobuf"
}
```
When `beforeMultiline = fold`:
```scala
object LL {
  val scalapbRuntime = "com.thesamet.scalapb" %% "scalapb-runtime" %
    scalapb.compiler.Version.scalapbVersion % "protobuf"
}
```
When `beforeMultiline = unfold`:
```scala
object LL {
  val scalapbRuntime =
    "com.thesamet.scalapb" %% "scalapb-runtime" % scalapb.compiler.Version.scalapbVersion %
      "protobuf"
}
```
With `fold` there is no newline after `=`, while with `unfold` newline is present.

Basically I would like to avoid newlines after `=`, but both settings can make them, and it seems like `fold` makes them more often.
